### PR TITLE
feat: support plugin-auth

### PR DESCRIPTION
**Pull request by mdonnalley**
### What does this PR do?
- add AuthRemover class
- add listAllAuthorizations method to AuthInfo
- add setAsDefault method to AuthInfo
- add setAlias method to AuthInfo

### What issues does this PR fix or reference?
@W-7978524@

### CODE REVIEW

File: src/authInfo.ts [hunk 1]
```diff
@@ -603,6 +659,44 @@ export class AuthInfo extends AsyncCreatable<AuthInfo.Options> {
...
+  public async setAsDefault(options: { defaultUsername?: boolean; defaultDevhubUsername?: boolean }) {
+    let config: Config;
+    // if we fail to create the local config, default to the global config
+    try {
+      config = await Config.create({ isGlobal: false });
+    } catch {
+      config = await Config.create({ isGlobal: true });
+    }
+
+    const value = ensureString(this.getFields().alias || this.getUsername());
```
Comment (amphro) [inline src/authInfo.ts:677]: Wait, `alias` is on the config fields?

File: src/authInfo.ts [hunk 2]
```diff
@@ -603,6 +659,44 @@ export class AuthInfo extends AsyncCreatable<AuthInfo.Options> {
...
+  }
+
+  /**
+   * Sets the provided alias to the username
+   * @param alias alias to set
+   */
+  public async setAlias(alias: string) {
+    const authFields = this.getFields();
+    await Aliases.parseAndUpdate([`${alias}=${authFields.username}`]);
+    const updated = Object.assign({}, authFields, { alias });
```
Comment (amphro) [inline src/authInfo.ts:696]: I didn't think we stored the alias in the stored file... that seems wrong to me.

On a separate note, doesn't `this.update({ alias })` do all the merging for you?

File: src/config/configAggregator.ts [hunk 3]
```diff
@@ -287,6 +287,9 @@ export class ConfigAggregator extends AsyncOptionalCreatable<JsonMap> {
    */
   private async loadProperties(): Promise<void> {
     this.resolveProperties(await this.globalConfig.read(), this.localConfig && (await this.localConfig.read()));
+    // const localConfig = (this.localConfig && await this.localConfig.exists()) ? await this.localConfig.read() : undefined;
```
Comment (amphro) [inline src/config/configAggregator.ts:290]: Remove?

File: src/authRemover.ts [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+ * const authConfigs = await remover.findAuthConfigs(
+ *  example@mycompany.com
+ * );
+ * const usernames = [...authConfigs.keys()];
+ * for (const username of usernames) {
+ *   await remover.removeAuth(username);
+ * }
+ * ```
+ */
+export class AuthRemover extends AsyncOptionalCreatable {
```
Comment (amphro) [inline src/authRemover.ts:46]: Nice job with this class. I like the API.

**Review by amphro — CHANGES_REQUESTED** [review]

File: src/authInfo.ts (hunk 2 above)
Comment (mdonnalley) [reply src/authInfo.ts:696]: @amphro It exists on the AuthFields interface (https://github.com/forcedotcom/sfdx-core/pull/294/files#diff-394ba1de20ae85e42959125d8ebc62b6R50) so it _could_ be there. However, I agree that the `Aliases` class should be the single source of truth for aliases. I'll change this accordingly

File: messages/auth.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "defaultOrgNotFound": "No %s org found",
+  "defaultOrgNotFoundAction": [
```
Comment (amphro) [inline messages/auth.json:3]: nit: Actions -< pluaral

File: src/authInfo.ts [hunk 6]
```diff
@@ -603,6 +659,44 @@ export class AuthInfo extends AsyncCreatable<AuthInfo.Options> {
...
+    await config.write();
+  }
+
+  /**
+   * Sets the provided alias to the username
+   * @param alias alias to set
+   */
+  public async setAlias(alias: string) {
+    const username = this.getUsername();
+    await Aliases.parseAndUpdate([`${alias}=${username}`]);
```
Comment (amphro) [inline src/authInfo.ts:697]: You could add a new method called `Aliases.update(alias: string, value: string)` if you want. This method was mainly to help with command line arguments but I'm also fine if you just leave it like this.

**Review by amphro — APPROVED** [review]

File: src/authRemover.ts (hunk 4 above)
Comment (shetzel) [reply src/authRemover.ts:51]: Yeah, this is awesome!

**Review by shetzel — APPROVED** [review]
👍

**Pull request merged**